API/CityApp-ANET/Services/CityService/CityService.cs

```diff
@@ -1,100 +1,94 @@
-            }
+    public async Task<List<CityDTO>> SearchCities(string name)
+    {
+        IQueryable<City> query = _context.Cities;
 
-            return await query.Select(c => DomainToDTO(c)).ToListAsync();
-        }
+        if (!string.IsNullOrEmpty(name)) query = query.Where(c => c.Name.ToLower().Contains(name.Trim().ToLower()));
 
-        public async Task<CityDTO?> GetCity(int id)
-        {
-            var city = await _context.Cities.FindAsync(id);
-            return city == null ? null : DomainToDTO(city);
-        }
+        return await query.Select(c => DomainToDTO(c)).ToListAsync();
+    }
 
-        public async Task<bool> PutCity(int id, City requestCity)
-        {
-            var city = await _context.Cities.FindAsync(id);
+    public async Task<CityDTO?> GetCity(int id)
+    {
+        var city = await _context.Cities.FindAsync(id);
+        return city == null ? null : DomainToDTO(city);
+    }
 
-            if (city == null)
-            {
-                return false;
-            }
+    public async Task<bool> PutCity(int id, CityDTO requestCity)
+    {
+        var city = await _context.Cities.FindAsync(id);
 
-            city.Name = requestCity.Name;
-            city.Photo = requestCity.Photo;
+        if (city == null) return false;
 
-            await _context.SaveChangesAsync();
+        city.Name = requestCity.Name;
+        city.Photo = requestCity.Photo;
 
-            return true;
-        }
+        await _context.SaveChangesAsync();
 
-        public async Task<CityDTO> PostCity(City requestCity)
-        {
-            _context.Cities.Add(requestCity);
-            await _context.SaveChangesAsync();
-            return DomainToDTO(requestCity);
-        }
+        return true;
+    }
 
-        public async Task<int?> DeleteCity(int id)
+    public async Task<CityDTO> PostCity(CityDTO requestCity)
+    {
+        var city = new City
         {
-            var city = await _context.Cities.FindAsync(id);
+            Name = requestCity.Name,
+            Photo = requestCity.Photo
+        };
 
-            if (city == null)
-            {
-                return null;
-            }
+        _context.Cities.Add(city);
 
-            _context.Cities.Remove(city);
+        await _context.SaveChangesAsync();
 
-            await _context.SaveChangesAsync();
+        return DomainToDTO(city);
+    }
 
-            return id;
-        }
+    public async Task<int?> DeleteCity(int id)
+    {
+        var city = await _context.Cities.FindAsync(id);
 
-        public CityDTO DomainToDTO(City city)
-        {
-            return new CityDTO
-            {
-                Id = city.Id,
-                Name = city.Name,
-                Photo = city.Photo
-            };
-        }
+        if (city == null) return null;
+
+        _context.Cities.Remove(city);
+
+        await _context.SaveChangesAsync();
+
+        return id;
     }
-}
 
+    public CityDTO DomainToDTO(City city)
```

Comment:
This function was removed from here and moved to `ICityService` with the most recently merged pull request. Did you do the rebase?